app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,8 @@
+class CommentsController < ApplicationController
+    
+    def create
+        @step = Step.find(params[:step_id])
+        @comment = @step.comments.create(comment_num: params[:comment_num],comment_text: params[:comment_text],step_id: params[:step_id],user_id: params[:user_id])
```

Comment:
I have solved the first issue of taking out step_id in create and understood why i removed it . The second one was in the beginning the created_at was not working and i had hardcoded the created_at time so all the comments have the same create time . But later created_at was working since i hardcoded it i was sorting it via comment num . So now as we discussed we have a helper which will sort the comments based on the created at attribute i will use this in the end .

Reply:
i have removed comment_num  and added sorting using created_at attribute !
DONE